### PR TITLE
patch revision endpoint added

### PR DESCRIPTION
**Pull request by MHekert**

### CODE REVIEW

File: src/controllers/revision.ts [hunk 1]
```diff
@@ -4,10 +4,12 @@ import isAuthorOfMode from '@controllers/middleware/isAuthorOfMode';
 import isUserLoggedIn from '@controllers/middleware/isUserLoggedIn';
 import addRevisionController from '@controllers/revision/addRevisionController';
 import getRevisionsController from '@controllers/revision/getRevisionsController';
+import patchRevisionController from '@controllers/revision/patchRevisionController';
```
Comment (JBudny) [inline src/controllers/revision.ts:8]: Names should be consistent throughout all files. You should change either the name of the imported variable in this file or in '@controllers/revision/patchRevisionController' to make it consistent. Check also the other similar files.

File: src/controllers/revision/patchRevisionController.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	validateRequest,
+	async (req: Request, res: Response, next: NextFunction) => {
+		try {
+			const {
+				body,
+				params: { modeId, revisionId }
+			} = req;
+			const updatedRevision = await updateRevision(modeId, revisionId, body);
+			if (!updatedRevision) return res.status(400).send(error400);
+			if (updatedRevision.nModified == 0) return res.status(507).send(allError507);
```
Comment (JBudny) [inline src/controllers/revision/patchRevisionController.ts]: You should probably use '===' operator here.

File: src/models/mode/helpers/updateRevision.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { isEmpty, pick } from 'lodash';
+
+import IRevision from '@interfaces/mode/IRevision';
+import Mode from '@models/mode/mode';
+
+const updateRevision = async (modeId: string, revisionId: string, changes: Partial<Omit<IRevision, 'createdAt'>>) => {
```
Comment (JBudny) [inline src/models/mode/helpers/updateRevision.ts:6]: That line is probably too long. I think you should try to set up tslint config max-line-length rule on something between 80-100

File: tests/controllers/revision/patchRevisionControllerTest.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+import createDummyModes from '@dummy/createDummyModes';
+import Mode from '@models/mode/mode';
+import app, { server } from '@src/server';
+import { error400 } from '@util/errorObjects';
+import { MONGODB_URI_TEST } from '@util/secrets';
+
+use(chaiHttp);
+const mongoUri: string = MONGODB_URI_TEST;
+const requestBody = { body: 'new body', code: 'newcode', changelog: ['new changelog1', 'new changelog2'] };
```
Comment (JBudny) [inline tests/controllers/revision/patchRevisionControllerTest.ts:13]: this line and L16 are probably too long

File: tests/controllers/revision/patchRevisionControllerTest.ts [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+	describe(`with correct params and empty body`, () => {
+		it(`should return status code 507`, async () => {
+			const modes = await createDummyModes();
+			const mode = modes.pop();
+			const revisionId = (<any>mode.revisions[0])._id;
+			const res = await request(app)
+				.patch(`/revision/${mode._id}/${revisionId}`)
+				.set('content-type', 'application/json');
+			expect(res).have.status(400);
```
Comment (JBudny) [inline tests/controllers/revision/patchRevisionControllerTest.ts:45]: In the description, you have typed the other error code

File: tests/controllers/revision/patchRevisionControllerTest.ts [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+	describe(`with wrong param`, () => {
+		it(`should return status code 507`, async () => {
+			const modes = await createDummyModes();
+			const mode = modes.pop();
+			const res = await request(app)
+				.patch(`/revision/${mode._id}/wrong_param`)
+				.set('content-type', 'application/json')
+				.send(requestBody);
+			expect(res).have.status(400);
```
Comment (JBudny) [inline tests/controllers/revision/patchRevisionControllerTest.ts:58]: In the description, you have typed the other error code

File: tests/controllers/revision/patchRevisionControllerTest.ts [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+	describe(`with correct params and wrong body`, () => {
+		it(`should return status code 200`, async () => {
+			const modes = await createDummyModes();
+			const mode = modes.pop();
+			const revisionId = (<any>mode.revisions[0])._id;
+			const res = await request(app)
+				.patch(`/revision/${mode._id}/${revisionId}`)
+				.set('content-type', 'application/json')
+				.send({ body: 2 });
+			expect(res).have.status(400);
```
Comment (JBudny) [inline tests/controllers/revision/patchRevisionControllerTest.ts:72]: In the description, you have typed the other error code

File: tests/models/mode/helpers/joinModeActionsTest.ts [hunk 8]
```diff
@@ -20,7 +20,8 @@ describe(`mode's model helper function joinModeAction`, () => {
 	before(async () => {
 		await connection.openUri(mongoUri, { useNewUrlParser: true, useCreateIndex: true });
 		await Promise.all([Mode.deleteMany({}), ModeAction.deleteMany({})]);
-		modesPromise = (async () => await Promise.all(createDummyModes()))();
+		// modesPromise = (async () => await Promise.all(createDummyModes()))();
```
Comment (JBudny) [inline tests/models/mode/helpers/joinModeActionsTest.ts]: Is that comment necessary here? If it's important I think it should be described

File: tests/models/mode/helpers/updateRevisionTest.ts [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+	after(async () => {
+		await Mode.deleteMany({});
+		return connection.close();
+	});
+
+	it('should correctly update revision with multiple changes', async () => {
+		const modes = await createDummyModes();
+		const mode = modes.pop();
+		const revisionId = (<any>mode.revisions[0])._id;
+		const changes = { body: 'new body', code: 'newcode', changelog: ['new changelog1', 'new changelog2'] };
```
Comment (JBudny) [inline tests/models/mode/helpers/updateRevisionTest.ts:24]: This line is probably a little too long

File: src/controllers/revision.ts (hunk 1 above)
Comment (MHekert) [reply src/controllers/revision.ts:8]: I will change it on different branch, for now it is at least consistent with other controllers.

File: src/models/mode/helpers/updateRevision.ts (hunk 3 above)
Comment (MHekert) [reply src/models/mode/helpers/updateRevision.ts:6]: For now all lines are shorter than 120 characters. You might be right that shorter lines would be more readable. I will change it in through all codebase on another branch.

**Review by JBudny — APPROVED** [review]

**Pull request merged**